db/db_impl/db_impl.cc

```diff
@@ -422,16 +414,31 @@ Status DBImpl::ResumeImpl(DBRecoverContext context) {
     // during previous error handling.
     if (file_deletion_disabled) {
       // Always return ok
-      s = EnableFileDeletions(/*force=*/true);
-      if (!s.ok()) {
+      Status enable_file_deletion_s = EnableFileDeletions(/*force=*/true);
+      if (!enable_file_deletion_s.ok()) {
         ROCKS_LOG_INFO(
             immutable_db_options_.info_log,
             "DB resume requested but could not enable file deletions [%s]",
-            s.ToString().c_str());
+            enable_file_deletion_s.ToString().c_str());
       }
     }
+  }
+
+  mutex_.Lock();
+  if (s.ok()) {
+    s = error_handler_.ClearBGError();
+  } else {
+    // NOTE: this is needed to pass ASSERT_STATUS_CHECKED
+    // in the DBSSTTest.DBWithMaxSpaceAllowedRandomized test.
+    // See https://github.com/facebook/rocksdb/pull/7715#issuecomment-754947952
+    error_handler_.GetRecoveryError().PermitUncheckedError();
+  }
+  mutex_.Unlock();
```

Comment:
I proposed to call s = error_handler_.ClearBGError(); after we know it's fine to wake up foreground, which I think is right before we LOG ROCKS_LOG_INFO(immutable_db_options_.info_log, "Successfully resumed DB");
- As the context, the orignal https://github.com/facebook/rocksdb/pull/3997 introducing DBImpl::Resume() calls s = error_handler_.ClearBGError(); very close to calling ROCKS_LOG_INFO(immutable_db_options_.info_log, "Successfully resumed DB"); while the later https://github.com/facebook/rocksdb/pull/6949 distances these two calls a bit.
- And it seems fine to me that s = error_handler_.ClearBGError(); happens after EnableFileDeletions(/*force=*/true); at least syntax-wise since these two functions are orthogonal. And it also seems okay to me that we re-enable file deletion before s = error_handler_.ClearBGError();, which basically is resetting some state variables.